mono/metadata/gc.c

```diff
@@ -221,7 +221,7 @@ mono_gc_run_finalize (void *obj, void *data)
 	runtime_invoke (o, NULL, &exc, NULL);
 
 	if (exc) {
-		/* fixme: do something useful */
+		mono_unhandled_exception(exc);
```

Comment:
Unfortunately, I have no idea... possibly they just didn't investigate what the corresponding .NET behaviour would be. I can go and try hunting it down in the commit history in mainline Mono if you want.